### PR TITLE
Linked Data Signatures for JWS

### PR DESCRIPTION
**Pull request by mprorock**
Per issue #45 - would love commentary or text adjustment from @OR13 and / or @selfissued as I know they also have desires for better JWS support

***
<a href="https://pr-preview.s3.amazonaws.com/mesur-io/vc-wg-charter/pull/47.html" title="Last updated on Feb 4, 2022, 2:35 PM UTC (b4bad81)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-wg-charter/47/f706a67...mesur-io:b4bad81.html" title="Last updated on Feb 4, 2022, 2:35 PM UTC (b4bad81)">Diff</a>

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

**Comment by csuwildcat** [comment]
I support this PR, and trust The Bank of @OR13's full faith and credit.

**Review by kdenhartog — APPROVED** [review]

File: index.html
Comment (selfissued) [inline index.html]: Add "will be provided" to the end of this sentence fragment so that it becomes a sentence.  Then I'd be glad to approve.

Comment (quartzjer) [reply index.html]: Ditto :+1:

**Comment by mprorock** [comment]
Thanks @selfissued - great catch

**Review by quartzjer — APPROVED** [review]

File: index.html
Comment (mprorock) [reply index.html]: should be in with 334d442 now

**Review by Sakurann — APPROVED** [review]

File: index.html
Comment (iherman) [inline index.html]:
```suggestion
                <a href="https://w3c-ccg.github.io/ldp-bbs2020/">BBS+ Cryptosuite</a>,
```

**Review by iherman — APPROVED** [review]

**Review by selfissued — APPROVED** [review]
Yes, we should do this.

**Review by brentzundel — APPROVED** [review]

**Comment by kdenhartog-mattr** [comment]
@msporny given the large amount of support of this PR I think we go ahead and merge it, but I thought you may object to this and wanted to make sure you're alright with this. We're going to wait 24 hours to see if you object otherwise we'll merge. If you object to this text being in there later because that wasn't enough time we can discuss changes for what can we agree to again later as well.

File: index.html
Comment (msporny) [inline index.html]:
```suggestion
Concrete serializations for <a href="https://github.com/w3c-ccg/lds-jws2020/">a JWS Cryptosuite</a> will be provided.
```

File: index.html
Comment (msporny) [inline index.html]:
```suggestion
                <a href="https://github.com/w3c-ccg/lds-jws2020/">JWS Cryptosuite</a>
```

File: index.html
Comment (msporny) [reply index.html]: We should merge this cryptosuite in with the others in the list above. There is no need to call it out any differently than the VC-JWT, Ed25519, and Secp256k1 ones.

**Review by msporny — APPROVED** [review]
I suggest that minor changes are made to bring it in line w/ all the other language changes in the charter (Linked Data -> Data Integrity).

**Comment by msporny** [comment]
> We're going to wait 24 hours to see if you object otherwise we'll merge.

As an aside, a 24 hour merge period for a PR that was raised yesterday is highly problematic. We've been giving VC WG Charter PRs /weeks/ before merging, even with lots of initial support. Again, this is not the merge process we've been following. 

Even though I agree with this PR being merged, giving /anyone/ in the WG 24 hours notice on a 1 day old PR is unacceptable.

**Comment by kdenhartog** [comment]
> Even though I agree with this PR being merged, giving /anyone/ in the WG 24 hours notice on a 1 day old PR is unacceptable.

Understood - the argument for merging so quickly was that we were making progress and it appeared to have overwhelming support in favor of this moving forward. Thank you for being able to get a review in on it today given it was a bit of an odd circumstance.

**Comment by msporny** [comment]
> the argument for merging so quickly was that we were making progress and it appeared to have overwhelming support in favor of this moving forward. 

For the record, that is a terrible argument, and that is not the process that we have consensus on in the WG.

The WG should know better than that. People can't make meetings for various reasons, minutes aren't released in those tight timeframes, people have a change of heart after thinking about something for a while... there are many reasons to not turn PRs around in 24 hours and this has /never/ been how the VCWG operates.

Case in point here is that none of the eight positive reviewers caught the fact that the PR was using outdated language that we've worked hard to excise from the charter.

Come on folks, we can do better than this from a process perspective.

File: index.html
Comment (selfissued) [reply index.html]: If we're going to change the title of the link, we should change it to the actual title, which is "JSON Web Signature 2020".

File: index.html
Comment (selfissued) [reply index.html]: If we're going to change the title associated with the link, we should use the correct one, which is "JSON Web Signature 2020".

As another point, it's not just "a JWS Cryptosuite".  It's a specific JWS cryptosuite - the one referenced by the link.

**Comment by brentzundel** [comment]
We will leave this PR open at least until our next WG call to give folks
more time to review it.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-01-26-vcwg#section7-3) on 2022-01-26

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 7.3. Linked Data Signatures for JWS (pr vc-wg-charter#47)

_See github pull request [vc-wg-charter#47](https://github.com/w3c/vc-wg-charter/pull/47)._




**Brent Zundel:** Related PR adds language to the charter.  
… Has been reviewed and approved..  
… Specifically this PR, is there language here that would be opposed?.  

> *Michael Jones:* Merge it!.

**Kyle Den Hartog:** I see no objectionable issues with this..  
… I think the one we need to check with is Manu.  
… I see has not approved it yet. That is one reason to hold off..  

**Brent Zundel:** Would you ask for him to look at it in the PR?.  

**Kyle Den Hartog:** Yes, he had some delays in his work, so we may be waiting, but I'll raise it.  

**Michael Jones:** I'm new to this WG and don't know your processes in general. But looking at it, there's 7 approvers. I don't think Manu has some special status over others in the WG. I would propose instead we merge it, and if he wants to create a PR with language modifying it, he could do so.  

> *Orie Steele:* +1 git will keep track of history, and we can debate PRs that have objections..

**Michael Jones:** I'd rather see progress than waiting.  

**Kyle Den Hartog:** Normally I would agree with you, but this PR has only been open a day.  
… I want to see if there will be controversy we can address ahead of time..  
… I can see arguments "It's already in there, don't pull it out".  
… Don't want to get into that too early.  
… Would like to get alignment and agreement. It's to manage the way the group is going to achieve consensus..  

> *Orie Steele:* +1 brent... this is a charter that still has to be approved....

**Brent Zundel:** Because this is just a draft charter, not a spec... If people actually showing up object to the work, it won't get approved.  
… So I have less heartburn about merging this PR....  

**Kyle Den Hartog:** In that case, I'm fine with merging it, if you feel comfortable that it's not going to be a problem. Mike makes a good point, let's make progress.  
… I'll just make sure Manu is at least alert to it.  

**Michael Jones:** Thanks, Kyle..  

**Joe Andrieu:** I don't understand the rush here. If there are some concerns that someone in this community might object, I think we should tag the and give them 24 hours to respond. That's all..  

> *Kristina Yasuda:* +1.

**Brent Zundel:** Opposition to course of action, Kyle will tag him, and if we don't see objections by tomorrow, I'll merge it?.  

> *Ted Thibodeau Jr.:* +1 to a tag and pause.

**Brent Zundel:** Thank you all for that conversation..  

</details>

**Comment by mprorock** [comment]
@msporny @selfissued - Just pushed a potential update to the language and list placement.  While i think this whole section needs a bit of work, i think this is more in line with what is being requested, and subsequent PRs could clean up formatting on the section.

File: index.html
Comment (mprorock) [reply index.html]: as above, please see ee77c9a

File: index.html
Comment (mprorock) [reply index.html]: do the adjustments in ee77c9a work?

**Comment by OR13** [comment]
We should merge this PR ASAP, so we can take up cleaning up the entire section: https://github.com/w3c/vc-wg-charter/issues/54

**Comment by iherman** [comment]
@brentzundel @Sakurann I let you decide on merging or not (both of you have the right to merge).

File: index.html
Comment (msporny) [reply index.html]: I was suggesting you be more general instead of locking yourself into that specific cryptosuite. If you lock yourself into that specific cryptosuite (which already has deployment) you might not be able to change things you want to. Again, helps to be a bit vague and state that you're taking an input document (and might change it significantly throughout the WGs existence). That is why all the other cryptosuites are not specific wrt. the output suite, other than it being a concrete one based on the input document.

**Comment by brentzundel** [comment]
@mprorock there seem to be merge conflicts. Are you able to resolve those? Then we can merge.

**Comment by mprorock** [comment]
@brentzundel conflicts should be resolved

**Comment by brentzundel** [comment]
multiple positive reviews, no stated objections

**Pull request merged**
